plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/geoip/GeoIpProcessor.java

```diff
@@ -68,13 +71,21 @@ public void execute(Data data) {
             throw new RuntimeException(e);
         }
 
-        final Map<String, Object> geoData;
+        Map<String, Object> geoData;
         switch (dbReader.getMetadata().getDatabaseType()) {
             case "GeoLite2-City":
-                geoData = retrieveCityGeoData(ipAddress);
+                try {
+                    geoData = retrieveCityGeoData(ipAddress);
+                } catch (Exception e) {
+                    geoData = Collections.singletonMap("error", ExceptionsHelper.detailedMessage(e));
```

Comment:
yes, we need a failure mechanism like in #14548.

but this is specifically for when the geodb has no record for an ip address which is common execution path.I think for this the geo processor itself should have defined logic that is separate from #14548